src/renderers/dom/fiber/ReactDOMFiberComponent.js

```diff
@@ -638,11 +648,19 @@ var ReactDOMFiberComponent = {
         break;
     }
 
-    assertValidProps(workInProgress, nextProps);
-    var isCustomComponentTag = isCustomComponent(workInProgress._tag, nextProps);
-    updateDOMProperties(workInProgress, rootContainerElement, lastProps, nextProps, isCustomComponentTag);
+    assertValidProps(tag, nextProps);
+    var wasCustomComponentTag = isCustomComponent(tag, lastProps);
+    var isCustomComponentTag = isCustomComponent(tag, nextProps);
```

Comment:
I think this is for the case where you switch from `<button is="my-custom-button" />` to `<button />`.